plugins/repository-s3/build.gradle

```diff
@@ -311,10 +313,12 @@ if (useFixture) {
     plugin file(tasks.bundlePlugin.archiveFile)
     environment 'AWS_CONTAINER_CREDENTIALS_FULL_URI', { "http://${s3Fixture.addressAndPort}/ecs_credentials_endpoint" }, IGNORE_VALUE
   }
-} else {
-  task s3ThirdPartyTests {
-    dependsOn check
+  s3ThirdPartyTests.doFirst {
+    throw new IllegalStateException("Tried to run third party tests but not all of the necessary environment variables 'amazon_s3_access_key', " +
```

Comment:
This won't work like we want since `s3ThirdPartyTests` doesn't actually do anything, it's just a lifecycle task that depends on the appropriate testing tasks we'll get failures before this actually runs. If there is a configuration issue we want to fail as early as possible, ideally, before we start executing _any_ tasks as it's annoying to get several minutes into a build only for it to fail because you didn't provide the right config.

Let's do `gradle.taskGraph.whenReady { if (it.hasTask(s3ThirdPartyTests)) { // throw exception} }` such that we throw this error if this task _will_ be executed. This error will happen during task graph calculation, so users will get an error before we actually start running anything.